src/o365/spfx/commands/project/project-rename.spec.ts

```diff
@@ -0,0 +1,413 @@
+    sinon.stub(fs, 'writeFileSync').callsFake(() => { });
+
+    cmdInstance.action({ options: { newName: 'spfx-react' } }, () => {
+      assert(trackEvent.called);
+    });
+  });
+
+  it('logs correct telemetry event', () => {
+    sinon.stub(command as any, 'getProjectRoot').callsFake(_ => path.join(process.cwd(), projectPath));
+    sinon.stub(fs, 'writeFileSync').callsFake(() => { });
+
+    cmdInstance.action({ options: { newName: 'spfx-react' } }, () => {
+      assert.equal(telemetry.name, commands.PROJECT_RENAME);
+    });
+  });
+
+  it('fails validation if newName is not passed', () => {
+    const actual = (command.validate() as CommandValidate)({ options: {} });
+    assert.notEqual(actual, true);
+  });
+
+  it('passes validation if newName is passed', () => {
+    const actual = (command.validate() as CommandValidate)({ options: { newName: 'spfx-react' } });
+    assert.equal(actual, true);
+  });
+
+  it('shows error if the project path couldn\'t be determined', (done) => {
+    sinon.stub(command as any, 'getProjectRoot').callsFake(_ => null);
+
+    cmdInstance.action({ options: { newName: 'spfx-react' } }, (err?: any) => {
+      try {
+        assert.equal(JSON.stringify(err), JSON.stringify(new CommandError(`Couldn't find project root folder`, 1)));
+        done();
+      }
+      catch (ex) {
+        done(ex);
+      }
+    });
+  });
+
+  it('replaces project name in package.json', (done) => {
+    sinon.stub(command as any, 'getProjectRoot').callsFake(_ => path.join(process.cwd(), projectPath));
+    let replacedContent = `{
+  "name": "spfx-react",
+  "version": "0.0.1",
+  "private": true,
+  "engines": {
+    "node": ">=0.10.0"
+  },
+  "scripts": {
+    "build": "gulp bundle",
+    "clean": "gulp clean",
+    "test": "gulp test"
+  },
+  "dependencies": {
+    "react": "16.7.0",
+    "react-dom": "16.7.0",
+    "@types/react": "16.7.22",
+    "@types/react-dom": "16.8.0",
+    "office-ui-fabric-react": "6.143.0",
+    "@microsoft/sp-core-library": "1.8.2",
+    "@microsoft/sp-property-pane": "1.8.2",
+    "@microsoft/sp-webpart-base": "1.8.2",
+    "@microsoft/sp-lodash-subset": "1.8.2",
+    "@microsoft/sp-office-ui-fabric-core": "1.8.2",
+    "@types/webpack-env": "1.13.1",
+    "@types/es6-promise": "0.0.33"
+  },
+  "resolutions": {
+    "@types/react": "16.7.22"
+  },
+  "devDependencies": {
+    "@microsoft/sp-build-web": "1.8.2",
+    "@microsoft/sp-tslint-rules": "1.8.2",
+    "@microsoft/sp-module-interfaces": "1.8.2",
+    "@microsoft/sp-webpart-workbench": "1.8.2",
+    "@microsoft/rush-stack-compiler-2.9": "0.7.7",
+    "gulp": "~3.9.1",
+    "@types/chai": "3.4.34",
+    "@types/mocha": "2.2.38",
+    "ajv": "~5.2.2"
+  }
+}`;
+
+    const writeFileSyncSpy: sinon.SinonStub = sinon.stub(fs, 'writeFileSync').callsFake(() => { });
+
+    cmdInstance.action({ options: { newName: 'spfx-react' } }, (err?: any) => {
+      try {
+        assert.equal(writeFileSyncSpy.getCall(0).args[1], replacedContent);
```

Comment:
Rather than expecting updating the file to be an n-th call, you could call `writeFileSyncSpy.calledWith` and pass the expected arguments. That way, should the implementation change and something else was written before, the order won't matter and it won't break the test

Reply:
Thanks. Getting used to Sinon! 😃